modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -1459,16 +1459,85 @@ private static boolean isListOrSet(CodegenParameter codegenParameter) {
     }
 
     private String replaceBeanValidationCollectionType(CodegenProperty codegenProperty, String dataType) {
-        if (!useBeanValidation() || !codegenProperty.isModel || isResponseType(codegenProperty)) {
+        if (!useBeanValidation() || isResponseType(codegenProperty)) {
             return dataType;
         }
 
         if (StringUtils.isEmpty( dataType ) || dataType.contains( "@Valid" )) {
             return dataType;
         }
-        return dataType.replace( "<", "<@Valid " );
+
+        if (codegenProperty.isModel) {
+            return dataType.replace( "<", "<@Valid " );
+        }
+        String beanValidation = getPrimitiveBeanValidation(codegenProperty);
+        if(beanValidation == null) {
```

Comment:
style nit: inconsistent spacing before `(`

Reply:
Iam not sure why this code style is not working with IntelliJ. Do you have hint for me ?

Reply:
Ah no I don't have any hints sorry 😔 

the formatting is not entirely consistent for this project either. I usually just try to match spacing and new line styles to the code immediately above and below myself.

Not a big deal either way